### DISCUSSION TITLE
optimizePackageImports not working Wildcart Imports from the barrel file

### DISCUSSION THREAD

**Discussion by RhythmAgg** [vercel/next.js#66408] [Help]
I am using `optimizePackageImports` in my nextJS project to eliminate barrel exports from my local barrel package. The package works well with named and default imports from the barrel file and only the required code is included in the bundle.

However if i do wildcart import from the barrel index.js, the entire module along with its dependencies are included in the stat object and have to be parsed further.

It looks like this in my case:
```
// page.js
import * as all from './barrel-package'
export default function Home() {
  return (
    <all.Test1 />
  )
 }
```
```
// barrel-package/index.js
import Image from 'next/image';
export { Test1 } from './test1'
export { Test2 } from './test2';
```
In my final bundle ( stat object ) the `Test2` and Image components are also included. On doing named import like
```
import {Test1} from './barrel-package'
```
Only the Test1 component is included in the bundle. So why does Image component get into the bundle with wildcart imports?

**Comment by benjaminho1998** [comment]
I'm running into this problem. Have you been able to find any solutions?

**Reply by RhythmAgg** [reply]
I actually wrote a codemod to eliminate these barrel files all together. Hence not using the optimizePackageImports anymore.
